### PR TITLE
FriendsEval class

### PR DESCRIPTION
**Pull request by winstonallo**
This is a big ass PR I apologize, I got carried away. If it is too much to review, I will split it into multiple smaller ones, just let me know:)

# Additions:
* FriendsEval class:
  * performs a simple Network Analysis on evaluations involving one user
  * shows the top 10 most involved with logins and gives user the option to see the full list
* Animation class:
  * moved full animation logic into the Animation class
  * start an animation by instantiating an object with your desired loading message: `loading_animation = Animation("<your message>")` 
  * stop the animation by calling `stop_animation()` on your Animation object
* Utils class: make_request_with_backoff():
  * created the make_request_with_backoff() function to adress the "429: Too Many Requests" response that the 42 API returns often
  * you can call it exactly like request.get(), the key difference is that on status code 429, it will backoff and retry
  * max_retries is set to 5 by default, I have found it to be enough but you can add it to your method call if you want to change it
 
Update: Merge conflicts with main are fixed, tested functionality & reformatted after fixes

### CODE REVIEW

**Review by ifaoji — CHANGES_REQUESTED** [review]
There seems to be a problem with the evaluation network analysis when looking at the full list. The list seems to be printed multiple times :thinking:

**Comment by winstonallo** [comment]
On it 👍🏻

**Comment by winstonallo** [comment]
[921fc29](https://github.com/42-stats/42-stats/pull/8/commits/921fc2947f1079b52653d14fbea92b109ec675f5)

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,6 @@ __pycache__
 
 # Editor
 .vscode
+
+# Tests
+*.json
```
Comment (ifaoji) [inline .gitignore]: Just for my mental health lol
```suggestion
*.json

```

File: src/CLInterface.py [hunk 2]
```diff
@@ -51,30 +53,23 @@ def prompt(self, options: list) -> str:
...
-
-        if request == "quit":
+        result = ""
+        if request in self.modules:
+            try:
+                result = self.modules[request].run()
+            except Exception as e:
+                self.error(f"\rerror: {e}")
+            self.show_result(result=result)
+        elif request == "quit":
```
Comment (ifaoji) [inline src/CLInterface.py]: Looks like the merging caused the old code to be used, could you update that to the new version?

**Review by ifaoji — CHANGES_REQUESTED** [review]

File: src/CLInterface.py (hunk 2 above)
Comment (winstonallo) [reply src/CLInterface.py]: you mean the one on main?

**Comment by winstonallo** [comment]
[a35729e](https://github.com/42-stats/42-stats/pull/8/commits/a35729ee2143a1c71b4093e48388840b8e2d1993)

**Review by ifaoji — APPROVED** [review]

**Pull request merged**